code/messaging/src/androidTest/java/com/adobe/marketing/mobile/messaging/E2EFunctionalTests.java

```diff
@@ -145,50 +163,147 @@ Map<String, Object> createExpectedEdgePersonalizationEventData() {
+    }
+
+    private void verifyInAppPropositionsRetrievedFromEdge() throws InterruptedException {
+        final Map<String, Object> expectedEdgePersonalizationEventData =
+                createExpectedEdgePersonalizationEventData();
         // verify message personalization request content event
         final List<Event> edgePersonalizationRequestEvents =
-                getDispatchedEventsWith(EventType.EDGE, EventSource.REQUEST_CONTENT);
+                getDispatchedEventsWith(EventType.EDGE, EventSource.REQUEST_CONTENT, 5000);
         assertEquals(1, edgePersonalizationRequestEvents.size());
         final Event edgePersonalizationRequestEvent = edgePersonalizationRequestEvents.get(0);
         assertEquals(
                 expectedEdgePersonalizationEventData,
                 edgePersonalizationRequestEvent.getEventData());
-        edgePersonalizationRequestEventID = edgePersonalizationRequestEvent.getUniqueIdentifier();
-
-        // verify edge personalization decision event
-        /* TODO will be fixed in feature/feed
-        final List<Event> edgePersonalizationDecisionsEvents = getDispatchedEventsWith(EventType.EDGE, MessagingConstants.EventSource.PERSONALIZATION_DECISIONS);
-        assertEquals(1, edgePersonalizationDecisionsEvents.size());
-        final Event edgePersonalizationDecisionEvent = edgePersonalizationDecisionsEvents.get(0);
-        final Map<String, Object> expectedInAppPayload = MessagingTestUtils.getMapFromFile("expectedInAppPayload.json");
-        final Map<String, Object> payload = (Map<String, Object>) ((List) edgePersonalizationDecisionEvent.getEventData().get("payload")).get(0);
-        final Map<String, Object> expectedScopeDetails = (Map<String, Object>) expectedInAppPayload.get("scopeDetails");
-        final Map<String, Object> scopeDetails = (Map<String, Object>) payload.get("scopeDetails");
-        assertEquals(expectedScopeDetails.get("activity"), scopeDetails.get("activity"));
-        assertEquals(expectedScopeDetails.get("correlationID"), scopeDetails.get("correlationID"));
-        assertEquals(expectedScopeDetails.get("decisionProvider"), scopeDetails.get("decisionProvider"));
-        assertEquals(expectedInAppPayload.get("scope"), payload.get("scope"));
+        final String edgePersonalizationRequestEventID =
+                edgePersonalizationRequestEvent.getUniqueIdentifier();
+
+        // verify personalization decisions event containing two in-app propositions
+        final List<Event> messagingPersonalizationEvents =
+                getDispatchedEventsWith(
+                        EventType.EDGE,
+                        MessagingConstants.EventSource.PERSONALIZATION_DECISIONS,
+                        3000);
+        final Event messagingPersonalizationEvent = messagingPersonalizationEvents.get(0);
+        assertEquals(
+                5, ((List) messagingPersonalizationEvent.getEventData().get("payload")).size());
```

Comment:
If we end up creating more campaigns this might fail. Do we need this assertion?

Reply:
good point, it can be removed